source/extensions/common/wasm/wasm.h

```diff
@@ -31,35 +31,36 @@ class Context;
 class Wasm;
 class WasmVm;
 
+// Represents a WASM-native word-sized datum. On 32-bit VMs, the high bits are always zero.
+// The WASM/VM API treats all bits as significant.
+struct Word {
```

Comment:
Actually, why can't we simply use `uint64_t`? It's effectively the same, isn't it? What's the benefit of wrapping it in a `struct`?

Reply:
We need to have something which the template magic can recognize and convert to a unint32_t or a unint64_t.   We also have hard uint32_t and hard uint64_t so we can't just reuse one or the other.